stdlib/functools.pyi

```diff
@@ -131,3 +133,19 @@ if sys.version_info >= (3, 8):
 
 if sys.version_info >= (3, 9):
     def cache(__user_function: Callable[..., _T]) -> _lru_cache_wrapper[_T]: ...
+
+class _HashedSeq(Generic[_T]):
+    __slots__: str
+    def __init__(self, tup: Tuple[Any, ...], hash: Callable[[object], int]  = ...) -> None: ...
+    def __hash__(self) -> int: ...
+
+def _make_key(
+    args: Hashable,
+    kwds: Hashable,
+    typed: bool,
+    kwd_mark: Optional[Tuple[object, ...]] = ...,
+    fasttypes: Optional[Set[type]] = ...,
+    tuple: Optional[Type[Tuple[Any, ...]]] = ...,
+    type: Optional[Any] = ...,
+    len: Optional[Callable[[Sized], int]] = ...,
+) -> Union[Dict[int, str], _HashedSeq[Any]]: ...
```

Comment:
This doesn't return dicts. In fact, the return type is always hashable:

```suggestion
) -> Union[Hashable, _HashedSeq[Any]]: ...
```

Union return types have their problems, but this might be fine for now. We can always change it later.

Reply:
To summarise, the user needs only to provide three arguments like this call:

```python
key = _make_key(args, kwargs, typed=False)
```

Reply:
That is why i've added Optional.

Reply:
Did you get me?